src/uint256.h

```diff
@@ -9,84 +9,70 @@
 #include <crypto/common.h>
 #include <span.h>
 
-#include <assert.h>
+#include <algorithm>
+#include <array>
+#include <cassert>
 #include <cstring>
 #include <stdint.h>
 #include <string>
-#include <vector>
 
 /** Template base class for fixed-sized opaque blobs. */
 template<unsigned int BITS>
 class base_blob
 {
 protected:
     static constexpr int WIDTH = BITS / 8;
-    uint8_t m_data[WIDTH];
+    std::array<uint8_t, WIDTH> m_data;
+    static_assert(WIDTH == sizeof(m_data), "Sanity check");
+
 public:
     /* construct 0 value by default */
     constexpr base_blob() : m_data() {}
 
     /* constructor for constants between 1 and 255 */
     constexpr explicit base_blob(uint8_t v) : m_data{v} {}
 
-    explicit base_blob(const std::vector<unsigned char>& vch);
+    constexpr explicit base_blob(Span<const unsigned char> vch)
+    {
+        assert(vch.size() == WIDTH);
+        std::copy(vch.begin(), vch.end(), m_data.begin());
+    }
 
-    bool IsNull() const
+    constexpr bool IsNull() const
     {
-        for (int i = 0; i < WIDTH; i++)
-            if (m_data[i] != 0)
-                return false;
-        return true;
+        return std::all_of(m_data.begin(), m_data.end(), [](uint8_t val) {
+            return val == 0;
+        });
     }
 
-    void SetNull()
+    constexpr void SetNull()
     {
-        memset(m_data, 0, sizeof(m_data));
+        std::fill(m_data.begin(), m_data.end(), 0);
     }
 
-    inline int Compare(const base_blob& other) const { return memcmp(m_data, other.m_data, sizeof(m_data)); }
+    constexpr int Compare(const base_blob& other) const { return std::memcmp(m_data.data(), other.m_data.data(), WIDTH); }
```

Comment:
nit: we've already defined member function `data()` to return `m_data.data()` (+ in a bunch more lines further down and in `uint256.cpp`)
```suggestion
    constexpr int Compare(const base_blob& other) const { return std::memcmp(data(), other.data(), WIDTH); }
```